### ISSUE TITLE
Textbox - VoiceOver does not read label with prefix prop

### ISSUE THREAD

**Issue by gusch** [Sage/carbon#7976]
### Description

When VoiceOver is active and i focus the second textbox, it does not read the label, only the prefix

### Reproduction

https://stackblitz.com/edit/parsium-carbon-starter-anqctqt7?file=src%2FApp.tsx,package.json

### Carbon version

159.3.1

### React version

v18

### Relevant browsers

Chrome

### Relevant OSs

MacOS

### Confidentiality

- [x] I confirm there is no confidential or commercially sensitive information included.

**Comment by nuria1110** [comment]
FE-7687
